importing/views.py

```diff
@@ -45,3 +64,157 @@ class DataImportDeleteView(CustomDeleteView):
+                            "station": "CAR_02_HC_01",
+                            "format": 47,
+                            "rawfile": "/media/imports/data_2024.csv",
+                            "visibility": "private",
+                            "reprocess": True,
+                            "date": "2024-12-09T10:30:00Z",
+                            "status": "N",
+                            "status_display": "Not queued",
+                        },
+                    )
+                ],
+            ),
+            400: OpenApiResponse(description="Invalid request parameters or file"),
+            403: OpenApiResponse(
+                description="User does not have permission to upload data."
+            ),
+            404: OpenApiResponse(description="Station or format not found"),
+        },
+        tags=["data-import"],
+    )
+    def post(self, request):
+        """Upload a data file and create a data import."""
+        # Get the uploaded file
+        rawfile = request.FILES.get("rawfile")
+        if not rawfile:
+            return Response(
+                {"rawfile": ["No file was submitted."]},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Build data dict for serializer
+        data = {
+            "station": request.data.get("station"),
+            "format": request.data.get("format"),
+            "visibility": request.data.get("visibility", "private"),
+            "reprocess": request.data.get("reprocess", False),
+            "observations": request.data.get("observations", ""),
+            "rawfile": rawfile,
+        }
+
+        # Validate request parameters
+        serializer = DataImportUploadRequestSerializer(data=data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        validated_data = serializer.validated_data
+        station = validated_data["station"]
+
+        # Check user permissions for this station
+        permitted_stations = self.get_permitted_stations(request.user)
+        if station not in permitted_stations:
+            return Response(
+                {
+                    "detail": "You do not have permission to upload data "
+                    f"for station '{station.station_code}'."
+                },
+                status=status.HTTP_403_FORBIDDEN,
+            )
+
+        # Create the data import
+        try:
+            data_import = DataImport.objects.create(
+                station=station,
+                format=validated_data["format"],
+                visibility=validated_data["visibility"],
+                reprocess=validated_data["reprocess"],
+                observations=validated_data.get("observations", ""),
+                rawfile=validated_data["rawfile"],
+                owner=request.user,
+            )
+        except Exception:
+            logging.exception("Error creating data import")
```

Comment:
In the logs, I would add the actual error from the exception, to be able to fix it, if needed.

```suggestion
        except Exception as e:
            logging.exception(f"Error creating data import: {e}")
```
Or something along these lines.